File: python/4.py

def is_pal(x):
    for i in range(len(x) // 2):
        if x[i] != x[-1 - i]:
            return False
    return True
M = 0
for i in range(999, 900, -1):
    for j in range(999, 900, -1):
        if is_pal(str(i * j)):
            if i * j > M:
                M = i * j

print(M)

File: cpp/4.cpp
#include <iostream>

using namespace std;

bool is_pal(string x){
    for (int i = 0; i < x.length() / 2; i++)
    {
        if (x[i] != x[x.length() - 1 - i])
        {
            return false;
        }
        
    }
    return true;
}


int main() {
    int M = 0;

    for (int i = 999; i > 900; --i)
    {
        for (int j = 999; j > 900; --j)
        {
            int N = i * j;
            if (is_pal(to_string(N)))
            {
                if (N > M)
                {
                    M = N;
                }
            }
            
        }
        
    }
    
    cout << M << endl;

    return 0;
}